eth/db/storage.py

```diff
@@ -185,11 +199,22 @@ def __init__(self, db: AtomicDatabaseAPI, storage_root: Hash32, address: Address
         big_endian encoding of the slot integer, and the rlp-encoded value.
         """
         self._address = address
-        self._storage_lookup = StorageLookup(db, storage_root, address)
+        self._journal_trie_locator = JournalDB(MemoryDB({TRIE_LOOKUP_KEY_NAME: storage_root}))
+
+        # One lookup for each starting trie. So far, this is only used to manage
+        # two tries: the original trie and a deleted one.
+        self._storage_lookups = {}
+        self._storage_lookups[storage_root] = StorageLookup(db, storage_root, address)
```

Comment:
I was just fooling around, but of course there are a lot of reasons this approach won't work. For example: the starting storage root is empty when the account doesn't exist at the beginning of the block. That will clash with the new "Deleted" trie at the empty storage root. Also, the storage_cache -- ... --> journal_storage are tied to the original storage lookup here, no matter which one is active. I can keep tweaking this, but... it's pretty messy and making me consider a larger refactor. I haven't seen a design that I'm happy with yet.